test/preprocess.spec.js

```diff
@@ -0,0 +1,66 @@
+/* global describe it */
+// @ts-check
+const chai = require('chai')
+const expect = chai.expect
+const dirtyChai = require('dirty-chai')
+
+chai.use(dirtyChai)
+
+const { preprocessVegaLite } = require('../src/preprocess.js')
+
+describe('preprocessVegaLite', () => {
+  it('should return original diagramText for invalid JSON', () => {
+    const invalidJson = 'invalid JSON'
+    expect(preprocessVegaLite(invalidJson, undefined)).to.be.equal(invalidJson)
+  })
+
+  it('should return original diagramText for valid JSON but without "data.url"', () => {
+    const validJsonWithoutDataUrl = '{}'
+    expect(preprocessVegaLite(validJsonWithoutDataUrl, undefined)).to.be.equal(validJsonWithoutDataUrl)
+  })
+
+  it('should return original diagramText for unexisting file referenced with "data.url"', () => {
+    const unexistingReferencedFile = `{
+      "data": {
+        "url": "unexisting.csv"
+      }
+    }`
+    expect(preprocessVegaLite(unexistingReferencedFile, {})).to.be.equal(unexistingReferencedFile)
+  })
+
+  it('should return diagramText with inlined local file referenced with "data.url"', () => {
+    const referencedLocalCsvFile = `{
+  "data": {
+    "url": "test/fixtures/vegalite-data.csv"
+  }
+}`
+    const inlinedLocalCsvFile = String.raw`{
```

Comment:
Any reason to add `String.raw` here?

Reply:
It lets me write `\n` instead of `\\n` for the `values` property, because escapes are not processed in `String.raw`. This is easier for me to grasp.

Reply:
Oh I see that's neat :ok_hand: